src/sentry/static/sentry/app/components/buttonBar.tsx

```diff
@@ -7,13 +8,41 @@ type ButtonBarProps = {
   className?: string;
   gap?: ValidSize;
   merged?: boolean;
+  active?: string | number;
   children: React.ReactNode;
 };
 
-function ButtonBar({children, className, merged = false, gap = 0}: ButtonBarProps) {
+function ButtonBar({
+  children,
+  className,
+  active,
+  merged = false,
+  gap = 0,
+}: ButtonBarProps) {
+  const shouldCheckActive = typeof active !== 'undefined';
   return (
     <ButtonGrid merged={merged} gap={gap} className={className}>
-      {children}
+      {!shouldCheckActive
+        ? children
+        : React.Children.map(children, child => {
+            if (!React.isValidElement(child)) {
+              return child;
+            }
+
+            const {props: childProps, ...childWithoutProps} = child;
+
+            // We do not want to pass `id` to <Button>`
+            const {id, ...props} = childProps;
+            const isActive = active === id;
+            const priority = isActive ? 'primary' : childProps.priority || 'default';
```

Comment:
Should the 'active' priority be an option on the button group and default to `primary`? We might want a button group where the active button is 'danger'.

Reply:
Yeah, I started off with it as a prop, but I wanted to just keep it simple for now, it would be trivial to add later on if we ever needed it. 

*edit* Oh, I forgot to carry the comment over on L42.